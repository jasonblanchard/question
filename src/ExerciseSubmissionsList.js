import React, { Component } from 'react';
import ScreenReaderContent from '@instructure/ui-core/lib/components/ScreenReaderContent'
import Table from '@instructure/ui-core/lib/components/Table';

import { average } from './utils/MathUtils';

class ExerciseSubmissionsList extends Component {
  render() {
    const submissions = this.props.exercise.submissions;
    const scores = submissions.map(submission => submission.score);
    const averageScore = average(scores);

    return (
      <div className="App">
        <h1>Submissions from Active Users</h1>
        <strong>Average Score</strong>: {averageScore}
        <Table striped="rows" caption={<ScreenReaderContent>submissions</ScreenReaderContent>}>
          <thead>
            <tr>
              <th>User Id</th>
              <th>Score</th>
            </tr>
          </thead>
          <tbody>
            {submissions.map(this._renderSubmission, this)}
          </tbody>
        </Table>
      </div>
    );
  }

  _renderSubmission(submission) {
    return (
      <tr key={submission.id}>
        <td>{submission.user.id}</td>
        <td>{submission.score}</td>
      </tr>
    );
  }
}

export default ExerciseSubmissionsList;
