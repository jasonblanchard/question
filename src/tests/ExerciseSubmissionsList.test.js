import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import ExerciseSubmissionsList from '../ExerciseSubmissionsList';

const exercise = {
  memberships: [
    {
      id: 1,
      status: 'ACTIVE',
      user: {
        id: 1
      }
    },
    {
      id: 2,
      status: 'ACTIVE',
      user: {
        id: 2
      }
    },
    {
      id: 3,
      status: 'INACTIVE',
      user: {
        id: 3
      }
    }
  ],
  submissions: [
    {
      id: 1,
      score: 100,
      user: {
        id: 1
      }
    },
    {
      id: 2,
      score: 80,
      user: {
        id: 2
      }
    },
    {
      id: 3,
      score: 70,
      user: {
        id: 3
      }
    }
  ]
};

it('renders with an Exercise', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<ExerciseSubmissionsList exercise={exercise} />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
