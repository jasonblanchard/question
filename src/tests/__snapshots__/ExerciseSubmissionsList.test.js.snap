// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with an Exercise 1`] = `
<div
  className="App"
>
  <h1>
    Submissions from Active Users
  </h1>
  <strong>
    Average Score
  </strong>
  : 
  83
  <Table
    caption={
      <ScreenReaderContent
        as="span"
      >
        submissions
      </ScreenReaderContent>
    }
    hover={false}
    size="medium"
    striped="rows"
  >
    <thead>
      <tr>
        <th>
          User Id
        </th>
        <th>
          Score
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          100
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          80
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          70
        </td>
      </tr>
    </tbody>
  </Table>
</div>
`;
